### PR TITLE
Add a SWIFT_STDLIB_STABLE_ABI flag to be able to build the stdlib without -enable-library-evolution

### PR DESCRIPTION
**Pull request by kubamracek**

### CODE REVIEW

**Comment by kubamracek** [comment]
The SWIFT_STDLIB_STABLE_ABI=0 setting should also be used on Linux, but I'm intentionally not flipping the default with this PR, and keeping it for a subsequent change (because it's likely to need some test adjustments).

File: CMakeLists.txt [hunk 1]
```diff
@@ -250,6 +250,10 @@ option(SWIFT_STDLIB_ENABLE_SIB_TARGETS
        "Should we generate sib targets for the stdlib or not?"
        FALSE)
 
+option(SWIFT_STDLIB_STABLE_ABI
+       "Should stdlib be built with stable ABI (library evolution, resilience)."
+       TRUE)
```
Comment (compnerd) [inline CMakeLists.txt:255]: Would you mind sinking this into `stdlib/CMakeLists.txt` please?  It's not used by the toolchain at all.

**Review by compnerd — APPROVED** [review]

**Comment by kubamracek** [comment]
@swift-ci please test

File: stdlib/cmake/modules/SwiftSource.cmake [hunk 2]
```diff
@@ -400,8 +400,8 @@ function(_compile_swift_files
     list(APPEND swift_flags "-Xfrontend" "-sil-verify-all")
   endif()
 
-  # The standard library and overlays are always built resiliently.
-  if(SWIFTFILE_IS_STDLIB)
+  # The standard library and overlays are built resiliently when SWIFT_STDLIB_STABLE_ABI=On.
+  if(SWIFTFILE_IS_STDLIB AND SWIFT_STDLIB_STABLE_ABI)
```
Comment (harlanhaskins) [inline stdlib/cmake/modules/SwiftSource.cmake:404]: Can you also make sure we don’t emit a module interface if library evolution is turned off?

**Comment by slavapestov** [comment]
> The SWIFT_STDLIB_STABLE_ABI=0 setting should also be used on Linux, but I'm intentionally not flipping the default with this PR, and keeping it for a subsequent change (because it's likely to need some test adjustments).

Why?

**Comment by gottesmm** [comment]
@kubamracek I think this is too strong of a condition to set a default of true. Consider porting swift to a new platform or that many of the platforms swift builds for do not have a stable ABI.

Also, we don't have a stable ABI on Linux today and do not want that to go into tree. So can we do that in this PR? At least with the swift-ci preset?

**Comment by gottesmm** [comment]
I am not sure what the right solution is btw with respect to how to determine the default value here. But TRUE makes me nervous.

**Comment by jckarter** [comment]
Turning library evolution off by default seems reasonable. I think that's good for setting expectations for new platforms, since establishing a stable ABI is a project to itself on top of doing the initial porting work.

**Comment by kubamracek** [comment]
I'm pretty sure we today build all platforms' stdlib with library evolution on. So it sounds like that as a preliminary step, we should turn it off for everything except Darwin?

File: stdlib/cmake/modules/SwiftSource.cmake (hunk 2 above)
Comment (kubamracek) [reply stdlib/cmake/modules/SwiftSource.cmake:407]: Done.

**Comment by kubamracek** [comment]
@swift-ci please test

**Comment by kubamracek** [comment]
Updated this PR to just introduce the "knob" to turn resilience on or off (SWIFT_STDLIB_STABLE_ABI). Made all currently supported platforms have the knob "on", even when it's technically not true, and marked those cases with a FIXME comment. Now we can separately work on turning the knob when applicable.

**Comment by kubamracek** [comment]
@swift-ci please test Windows platform

**Comment by kubamracek** [comment]
@harlanhaskins @jckarter @gottesmm looks good now?

**Comment by jckarter** [comment]
LGTM!

**Pull request merged**